### ISSUE TITLE
Installation for MacPort and Windows

### ISSUE THREAD

**Issue by yasu-sh** [pygraphviz/pygraphviz#477]
I hope this instruction in manual is updated.
https://pygraphviz.github.io/documentation/stable/install.html

This does not work on windows.
```
python -m pip install --use-pep517 `
              --config-setting="--global-option=build_ext" `
              --config-setting="--global-option="-IC:\Program Files\Graphviz\include" `
              --config-setting="--global-option="-LC:\Program Files\Graphviz\lib" `
              pygraphviz
```

This did work on windows.            
```
python -m pip install --use-pep517 `
              --config-setting="--global-option=build_ext" `
              --config-setting="--global-option=-IC:\Program Files\Graphviz\include" `
              --config-setting="--global-option=-LC:\Program Files\Graphviz\lib" `
              pygraphviz
```
Comments
https://github.com/pygraphviz/pygraphviz/commit/b5d6453ec22f2c23c0a9c1fed1bc049e391b2933#commitcomment-127008521
              
![image](https://github.com/pygraphviz/pygraphviz/assets/89838961/294c8b46-c905-434b-bb46-1de96b63b6f1)

**Comment by yasu-sh** [comment]
I think this modification will be applied the same as #468.
MacPorts / Chocolately. #468 covers only Powershell code. Chololately, MacPorts should be done after checking.

**Comment by NauticalMile64** [comment]
What happened when you tried to install using this method? I get a bunch of linker errors when following this procedure (graphviz 2.46.0 (20210118.1747), Python 3.11.4, and Visual Studio 2022). I can post the errors here or open a new issue if it is a separate problem.

**Comment by yasu-sh** [comment]
@NauticalMile64 I did not have linker error. Before I install C++ build tools, I might have a linker error.
You may make another issue on your problem. 
```powershell
PS E:> python -m pip install --use-pep517 `
               --config-setting="--global-option=build_ext" `
               --config-setting="--global-option="-IC:\Program Files\Graphviz\include" `
               --config-setting="--global-option="-LC:\Program Files\Graphviz\lib" `
               pygraphviz
ERROR: Invalid requirement: 'Files\\Graphviz\\include \n              --config-setting=--global-option=-LC:\\Pro
gram Files\\Graphviz\\lib'
Hint: It looks like a path. File 'Files\Graphviz\include
              --config-setting=--global-option=-LC:\Program Files\Graphviz\lib' does not exist.
```

- Graphviz version
```powershell
PS C:\Program Files\Graphviz\bin> dot -V
dot - graphviz version 8.1.0 (20230707.0739)
```

- python version
```powershell
PS E:> python -V
Python 3.10.10
```

- Microsoft Build Tools  https://visualstudio.microsoft.com/visual-cpp-build-tools/ (Not VS 2022)
```powershell
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools>cl
Microsoft(R) C/C++ Optimizing Compiler Version 19.37.32822 for x64
```
This is written in the page of [installation page of pygraphviz](https://pygraphviz.github.io/documentation/stable/install.html#windows-install:~:text=You%20may%20also%20need%20to%20install%20Visual%20C/C%2B%2B%2C%20e.g.%20from%20here%3A)

**Comment by NauticalMile64** [comment]
TLDR; To install on Windows 10, make sure VS is up-to-date, and use "x64 Native Tools Command Prompt for VS 2022" in Administrator Mode. The final command I used to successfully install is
```
python -m pip install --use-pep517 --config-setting="--global-option=build_ext" --config-setting="--global-option=-IC:\Program Files\Graphviz\include" --config-setting="--global-option=-LC:\Program Files\Graphviz\lib" pygraphviz
```
***
I was able to resolve my issue and install on my Windows 10 system. My PC rebooted to install some updates last night, so that may have updated some Visual Studio stuff in the background.

The other difference that could have made it work is that I used a different command prompt. When I was trying yesterday, I used the prompt called "Developer Command Prompt for Visual Studio 2022". This morning, I successfully installed it using the "x64 Native Tools Command Prompt for VS 2022". By the way I always run in Administrator Mode.

For the `pip` command, I removed the back ticks (\`) from the command and removed the line breaks. I also removed the second set of quotation marks from the second `--config-setting` options. Before I made that change, I was getting `Invalid Requirement` error messages similar to @yasu-sh (on Power Shell and on "Developer Command Prompt for Visual Studio 2022").

I used the same command on "Developer Command Prompt for Visual Studio 2022", which found `cl`, but triggered the linker errors. So fixing the command wasn't the only issue.

I tested the installation using the [Subgraph Example](https://pygraphviz.github.io/documentation/stable/auto_examples/plot_subgraph.html#sphx-glr-auto-examples-plot-subgraph-py).

**Comment by PinkShnack** [comment]
I used this command on Windows 10. Tested and worked on python 3.11 and python 3.10. I updated my Visual Studio, but the above commands didn't work for me.

```python
pip install --global-option=build_ext --global-option="-IC:\Program Files\Graphviz\include" --global-option="-LC:\Program Files\Graphviz\lib" pygraphviz
```

Taken from top answer on the this [stack overflow question](https://stackoverflow.com/questions/59707234/issues-installing-pygrahviz-fatal-error-c1083-cannot-open-include-file-graph).

**Comment by yasu-sh** [comment]
@PinkShnack As for python/graphviz, 3 years looks considerably old information, unfortunately.
 
Have you checked the options? like Desktop development with C++ at Visual Studio? Some Windows SDK might need to be installed. Error messages says many implies.  
![image](https://github.com/pygraphviz/pygraphviz/assets/89838961/4272cb10-6414-4916-b9ae-847e3e4540f4)

**Comment by yasu-sh** [comment]
@PinkShnack Thanks for you infos. I understand.
